Generate boot-debug.img in non-GKI config

GangSecurity opened the pull request:
For Generic Kernel Image (GKI) using devices that have a vendor_boot partition, boot-debug.img mustn't be flashed, as the boot partition must be flashed with a certified GKI image.
Instead vendor_boot-debug.img should be flashed onto the vendor_boot partition in order to facilitate debug ramdisk.

Test Done:
Boot

Traked-On: OAM-132990

GangSecurity committed:
Generate boot-debug.img in non-GKI config

For Generic Kernel Image (GKI) using devices that have a vendor_boot
partition, boot-debug.img mustn't be flashed, as the boot partition
must be flashed with a certified GKI image.
Instead vendor_boot-debug.img should be flashed onto the vendor_boot
partition in order to facilitate debug ramdisk.

Test Done:
Boot

Tracked-On: OAM-132990
Signed-off-by: Gang G Chen <gang.g.chen@intel.com>

sysopenci commented:
Improper Commit Message
Tracked on not found in commit message,
    make sure Tracked-On: Jira-ticket  is present.

GangSecurity committed:
Add init_boot to GPT images

boot-debug.img will not be generate if enables init_boot image

Test Done:
Boot

Tracked-On: OAM-132990
Signed-off-by: Gang G Chen <gang.g.chen@intel.com>

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has completed MERGE Build for this pr, build is SUCCESS. Please check the linked Tracked-On issue/Android CI Web  for more details. For Binaries: /android-ci-local/celadon-merge/1283
